plugin/advancedsubscription/src/AdvancedSubscriptionPlugin.php

```diff
@@ -0,0 +1,872 @@
+     * @param int $sessionId
+     * @param bool $save
+     * @return bool|int
+     */
+    public function sendMailMessage($studentId, $receiverId, $subject, $content, $sessionId, $save = false)
+    {
+        global $_configuration; // @TODO: Add $_configuration['no_reply_user_id'] to configuration file
+
+        $mailId = MessageManager::send_message(
+            $receiverId,
+            $subject,
+            $content,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            $_configuration['no_reply_user_id']
+        );
+
+        if ($save && !empty($mailId)) {
+            // Save as sent message
+            $this->saveLastMessage($mailId, $studentId, $sessionId);
+        }
+        return $mailId;
+    }
+
+    /**
+     * Check if session is open for subscription
+     * @param $sessionId
+     * @param string $fieldVariable
+     * @return bool
+     */
+    public function isSessionOpen($sessionId, $fieldVariable = 'es_abierta')
+    {
+        $sessionId = (int) $sessionId;
+        $fieldVariable = Database::escape_string($fieldVariable);
+        $isOpen = false;
+        if ($sessionId > 0 && !empty($fieldVariable)) {
+            $sfTable = Database::get_main_table(TABLE_MAIN_SESSION_FIELD);
+            $sfvTable = Database::get_main_table(TABLE_MAIN_SESSION_FIELD_VALUES);
+            $joinTable = $sfvTable . ' sfv INNER JOIN ' . $sfTable . ' sf ON sfv.field_id = sf.id ';
+            $row = Database::select(
+                'sfv.field_value as field_value',
+                $joinTable,
+                array(
+                    'where' => array(
+                        'sfv.session_id = ? AND sf.field_variable = ?' => array($sessionId, $fieldVariable),
+                    )
+                )
+            );
+            if (isset($row[0]) && is_array($row[0])) {
+                $isOpen = (bool) $row[0]['field_value'];
+            }
+        }
+
+        return $isOpen;
+    }
+
+    /**
+     * Update the queue status for subscription approval rejected or accepted
+     * @param $params
+     * @param $newStatus
+     * @return bool
+     */
+    public function updateQueueStatus($params, $newStatus)
+    {
+        if (isset($params['queue']['id'])) {
+            $where = array(
+                'id = ?' => $params['queue']['id'],
+            );
+        } elseif(isset($params['u']) && isset($params['s'])) {
+            $where = array(
+                'user_id = ? AND session_id = ? AND status <> ? AND status <> ?' => array(
+                    $params['u'],
+                    $params['s'],
+                    $newStatus,
+                    ADV_SUB_QUEUE_STATUS_ADMIN_APPROVED,
+                ),
+            );
+        }
+        if (isset($where)) {
+            $res = (bool) Database::update(
+                Database::get_main_table(TABLE_ADV_SUB_QUEUE),
+                array(
+                    'status' => $newStatus,
```

Comment:
$newStatus should already be filtered when you attend the comment above, but just in case...